app/models/problem_check.rb

```diff
@@ -1,6 +1,26 @@
 # frozen_string_literal: true
 
 class ProblemCheck
+  class Collection
+    include Enumerable
+
+    def initialize(checks)
+      @checks = checks
+    end
+
+    def each(...)
+      checks.each(...)
+    end
+
+    def run
```

Comment:
I think it may help a bit with readability for this to be called `run_all`, then on the dashboard controller you would get `ProblemCheck.realtime.run_all`; `ProblemCheck.realtime.run` makes it seem to my eyes like it's just a single check.

Reply:
I could live with that. 👍